packages/dcos-integration-test/extra/test_mesos.py

```diff
@@ -157,7 +158,11 @@ def test_if_ucr_app_runs_in_new_pid_namespace(dcos_api_session):
         marathon_framework_id = dcos_api_session.marathon.get('/v2/info').json()['frameworkId']
         app_task = dcos_api_session.marathon.get('/v2/apps/{}/tasks'.format(app['id'])).json()['tasks'][0]
 
-        content = dcos_api_session.mesos_sandbox_file(
-            app_task['slaveId'], marathon_framework_id, app_task['id'], ps_output_file)
+        # There is a race between the `app_task` starting and it writing its output to the `ps_output_file`.
```

Comment:
Yeah, I guess `race` was a poor choice of words. The underlying problem is that the current test assumes that the file is there immediately after the app reports that it is alive. This is a bad assumption because the app needs time to write the file before the test can read it.